src/KinesisVideoProducer.cpp

```diff
@@ -194,7 +195,9 @@ KinesisVideoProducer::~KinesisVideoProducer() {
 
 KinesisVideoProducerMetrics KinesisVideoProducer::getMetrics() const {
     STATUS status = ::getKinesisVideoMetrics(client_handle_, (PClientMetrics) client_metrics_.getRawMetrics());
-    LOG_AND_THROW_IF(STATUS_FAILED(status), "Failed to get producer metrics with: " << status);
+    if(STATUS_FAILED(status)) {
+        LOG_ERROR("Failed to get producer metrics with: " << status);
```

Comment:
This is a behavioral change and a dangerous one too. Understandably, this is just getting metrics but it might be worth having a way for the caller to distinguish a failed call.